src/components/addBook/AddBook.js

```diff
@@ -1,46 +1,53 @@
-import React, { useState } from 'react';
-import { useDispatch } from 'react-redux';
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import uuid from 'react-uuid';
 import swal from 'sweetalert';
-import { addBook } from '../../redux/books/books';
+import { createBook, fetchBooks, getBooksStatus } from '../../features/book/bookSlice';
 import './AddBook.css';
 
 const AddBook = () => {
   const dispatch = useDispatch();
+  const { status } = useSelector(getBooksStatus);
   const [book, setBook] = useState({
-    id: '',
+    item_id: '',
     title: '',
     author: '',
-    genre: 'Action',
-    percentage: 66,
+    category: 'Fiction',
+    percentage: 0,
     completed: false,
-    currentChapter: 'Chapter 17',
+    currentChapter: '',
   });
 
   const handleChange = (e) => {
     const { name } = e.target;
     setBook({
       ...book,
-      id: uuid(),
+      item_id: uuid(),
       [name]: e.target.value,
-      genre: 'Action',
+      category: 'Action',
       percentage: 66,
       completed: false,
       currentChapter: 'Chapter 17',
-
     });
   };
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    dispatch(addBook(book));
-    swal('Success', `"${book.title}" added Successfully`, 'success');
+    dispatch(createBook(book));
+    swal('Success', `"${book.title}" added Successfully`, 'success').then(() => {
+    });
     setBook({
       title: '',
       author: '',
     });
   };
 
+  useEffect(() => {
+    if (status === 'idle') {
+      dispatch(fetchBooks());
+    }
+  }, [status, dispatch]);
+
```

Comment:
- [ ] Great job on working on the functionality but I think there is a bug when adding a new book so after adding a book I can't see it till I refresh the page so each time I add a new book I should refresh the page so kindly fix it to add without refreshing ✨💯🚀🚀

Reply:
@Reem-lab thank you for your review. As a matter of fact, I really don't know what to try again😩. I knew the problem but I had to submit it anyways. Would you happen to have an idea about how I can go about it?
The API doesn't return the rest of the books and so I don't know how to update the state.
Thank you🥺

Reply:
@Reem-lab thank you. Changes has been made